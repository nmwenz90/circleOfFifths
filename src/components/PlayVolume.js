
import React from 'react';



function PlayVolume({ playSequence, stopSequence, undo }) {

    return (
        <div>
            <button onClick={playSequence}>Play</button>
            <button onClick={stopSequence}>Stop</button>
            <button onClick={undo}>Undo</button>
        </div>
    )
}

export default PlayVolume;

