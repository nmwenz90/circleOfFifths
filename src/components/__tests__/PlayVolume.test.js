import React from 'react';
import { shallow, mount } from 'enzyme';
import PlayVolume from '../PlayVolume';


describe('PlayVolume', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<PlayVolume debug />);

        expect(component).toMatchSnapshot();
    });
    it('should run the a function when play button is clicked', () => {
        let clickFn = jest.fn()
        const component = shallow(<button id="play" onClick={clickFn} >Icon</button>)
        component.find({ id: 'play' }).simulate('click')
        expect(clickFn).toHaveBeenCalled();
    })
    it('should run the a function when stop button is clicked', () => {
        let clickFn = jest.fn()
        const component = shallow(<button id="stop" onClick={clickFn} >Icon</button>)
        component.find({ id: 'stop' }).simulate('click')
        expect(clickFn).toHaveBeenCalled();
    })
    it('should run the a function when undo button is clicked', () => {
        let clickFn = jest.fn()
        const component = shallow(<button id="undo" onClick={clickFn} >Icon</button>)
        component.find({ id: 'undo' }).simulate('click')
        expect(clickFn).toHaveBeenCalled();
    })
});