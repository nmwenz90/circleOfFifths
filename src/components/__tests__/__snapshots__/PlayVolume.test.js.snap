// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayVolume should render correctly in "debug" mode 1`] = `
<div
  className="columns"
>
  <button
    className="button is-rounded is-dark"
    id="playButton"
  >
    <i
      className="fas fa-play-circle"
    />
  </button>
  <button
    className="button is-rounded is-dark"
  >
    <i
      class="fas fa-stop-circle"
    />
  </button>
  <button
    className="button is-rounded is-dark"
  >
    <i
      class="fas fa-undo-alt"
    />
  </button>
</div>
`;
